handlers/handlers.go

```diff
@@ -0,0 +1,197 @@
+package handlers
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/codingpop/simplerest/db"
+	"github.com/go-chi/chi"
+	"github.com/jackc/pgx/v4"
+)
+
+// Handlers ...
+type Handlers struct {
+	db *db.DB
+}
+
+// New creates a new instance of Handlers
+func New(db *db.DB) *Handlers {
+	return &Handlers{
+		db: db,
+	}
+}
+
+// GetPosts ...
+func (h *Handlers) GetPosts(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	posts, err := h.db.GetPosts()
+	if err != nil {
+		log.Println(err)
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+		return
+	}
+
+	if err := json.NewEncoder(w).Encode(map[string][]db.Post{"posts": posts}); err != nil {
+		http.Error(w, "failed to write response", http.StatusInternalServerError)
+	}
+}
+
+// GetPost ...
+func (h *Handlers) GetPost(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	id, err := strconv.Atoi(chi.URLParam(r, "id"))
+	if err != nil {
+		log.Println(err)
+
+		http.Error(w, "invalid id", http.StatusBadRequest)
+		return
+	}
+
+	post, err := h.db.GetPost(id)
+	if err != nil {
+		log.Println(err)
+
+		if err == pgx.ErrNoRows {
+			http.Error(w, "post not found", http.StatusNotFound)
+			return
+		}
+
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+		return
+	}
+
+	resp := map[string]db.Post{"post": post}
+
+	if err := json.NewEncoder(w).Encode(resp); err != nil {
+		log.Println(err)
+
+		http.Error(w, "failed to write response", http.StatusInternalServerError)
+	}
+}
+
+// UpdatePost ..
+func (h *Handlers) UpdatePost(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	var payload struct {
+		Title string `json:"title"`
+		Body  string `json:"body"`
+	}
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&payload); err != nil {
+		log.Println(err)
+
+		http.Error(w, "invalid request body", http.StatusBadRequest)
+		return
+	}
+
+	id, err := strconv.Atoi(chi.URLParam(r, "id"))
+	if err != nil {
+		log.Println(err)
+
+		http.Error(w, "invalid id", http.StatusBadRequest)
+		return
+	}
```

Comment:
Small detail, it may be better to move this above the JSON decode, since it's a cheaper check, and if it's wrong you don't have to do the JSON decode.